### ISSUE TITLE
glib fails to build for aarch64-linux

### ISSUE THREAD

**Issue by very-good-fish** [NixOS/nixpkgs#203363]
### Steps To Reproduce
1. build *glib* for `aarch64-linux` with `nix-build -E '(import <nixpkgs> { crossSystem = "aarch64-linux" ; }).glib'`

### Build log
[gist](https://gist.github.com/very-good-fish/67c64f19c49c9166ab14e7a65ea50eb4)

### Notify maintainers
@bobby285271 @hedning @dasj19 @maxeaubrey @lovek323 @7c6f434c

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.76-1.fc32.qubes.x86_64, Debian GNU/Linux, 11 (bullseye), nobuild`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.7`
 - channels(user): `"nixos-22.05.4338.695b3515251, nixpkgs-23.05pre429856.b45ec953794"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/user/channels/nixpkgs`
```

**Comment by wegank** [comment]
The problem does not occur on my aarch64-linux machine. Looks like it's cross-compiling only?

**Comment by bobby285271** [comment]
> `meson.build:2413:2: ERROR: Program 'xsltproc' not found or not executable`

Duplicate of #201994

- Fix for unstable: http://nixpk.gs/pr-tracker.html?pr=200760
- Fix for 22.11: http://nixpk.gs/pr-tracker.html?pr=202257
